libs/ui/avatar/brain/src/lib/fallback/brn-avatar-fallback.directive.ts

```diff
@@ -8,20 +8,11 @@ import type { ClassValue } from 'clsx';
 })
 export class BrnAvatarFallbackDirective {
 	private readonly element = inject(ElementRef).nativeElement;
-	readonly userCls = signal<ClassValue>('');
-	readonly useAutoColor = signal(false);
-	readonly textContent = inject(ElementRef).nativeElement.textContent;
+
+	public readonly userCls = input<ClassValue>('', { alias: 'class' });
+	public readonly useAutoColor = input(false, { alias: 'autoColor', transform: booleanAttribute });
```

Comment:
I saw the `NumberInput` one but didn't know about `BooleanInput`. I'll update this! 

I agree that the alias is unnecessary now, but the existing names are being used in other places and could be seen as a breaking change so I didn't rename it. Should I rename it anyway? We are still "alpha" so it should be fine?

Reply:
Thank you!

Yes, so we have identified the list of issues we want to get completed as part of the lead up to our v1 release, and we have identified signal migration as a place where breaking changes are likely to occur, and this has proven to be the case in the few PRs for signal migrations that have arisen.

In my opinion we are still in alpha, so now is the time for us to get the codebase in the best shape before v1 as breaking changes after that point are a much bigger issue. Having unneeded aliasing and other workarounds etc, is going to clutter the code base and make it less maintainable going forward.

We have seen similar cases in #447 and #449, and I suspect there will be more too.

That being said, breaking changes are not ideal, and we should minimize there impact on consumers. I do think it could be worth having an upgrade guide detailing all the breaking changes and showing the fix, in this case, it would be a rename, but in the cases like button, it may be slightly more involved. We have a Changelog page on the docs which is one place where this kind of content could be put, but I'm open to suggestions.

Would be keen to hear your thoughts on how we handle these breaking changes that are going to arise @thatsamsonkid, @goetzrobin, @elite-benni. I think we should come to a consensus so we know how to approach this going forward.

- Do we make the breaking changes now to improve the codebase while we are still in alpha?
- If so should we document it? And where?

Reply:
minor: Similarly I've retained `userCls` here and in another MR instead of renaming it to `userClass` for consistency, but we should just change that too.

I've seen the same issue for inputs that you've fixed with the provide pattern in some other components like Sheet & Hover Card. Those will need to be breaking changes anyway, so I feel it's fine to have renaming changes.

Reply:
I've renamed `useAutoColor` to `autoColor` (with `BooleanInput`), `userCls` to  `userClass`, and updated their references + template (manually).

Reply:
I've added the breaking changes info in the commit message following commitizen format, so we could potentially generate a changelog from them (if we are not already doing that?)

Reply:
@thatsamsonkid @goetzrobin @ashley-hunter @ajitzero 
I started a discussion for the breaking change topic.

Reply:
I'm resolving this thread as I've addressed the feedback points with new commits.